pkg/controllers/dynakube/extension/eec/reconciler.go

```diff
@@ -70,3 +86,100 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 
 	return r.createOrUpdateStatefulset(ctx)
 }
+
+func (r *reconciler) reconcileTlsSecret(ctx context.Context) error {
+	query := k8ssecret.Query(r.client, r.client, log)
+
+	if r.dk.GetExtensionsTlsRefName() != "" {
```

Comment:
can this be moved to properties and renamed so it's clearer what this does, like `NeedsSelfSignedTls()`?

Reply:
Yes, I thought the same, on it